QModManager/Error.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using UnityEngine.UI;
+
+namespace QModManager
+{
+    internal static class Error
+    {
+        internal static void ShowError(string error, Action onLeftButtonPressed = null, Action onRightButtonPressed = null, string leftButtonText = "See Log", string rightButtonText = "Close")
```

Comment:
Optional parameters of the same type can make for messy calls of the method in question.  
You may often find that you will have to constantly add the parameter names to make this work.

**Proposal:**  
Create a new abstract class `ButtonAction` that exposes a `string Text` property and a `Action ButtonPressed` property.  
Then, you can create `LeftButtonAction : ButtonAction` and `RightButtonAction : ButtonAction` classes and use them as optional parameters here with `null` as their default.  
```CSharp
ShowError(string error, LeftButtonAction leftButton = null, RightButtonAction rightButton = null);
```
This way, there is no ambiguity in the call and no need to use parameter names to clearly express which text and which action belong to which button.

Reply:
I personally don't see a point in bundling two parameters in a class, as 3 out of the 4 optional parameters will never be used in the release buld - only for testing